tests/avro/test_message_serializer.py

```diff
@@ -80,5 +81,23 @@ def test_decode_none(self):
 
         self.assertIsNone(self.ms.decode_message(None))
 
+    def test__get_subject_for_key(self):
+        basic = avro.loads(data_gen.BASIC_SCHEMA)
+        topic = "topic"
+        self.ms.registry_client.key_subject_name_strategy_func = topic_record_name_strategy  # noqa
+        subject = self.ms._get_subject(topic=topic, schema=basic, is_key=True)
+
+        expected = "topic-basic-key"
+        self.assertEqual(expected, subject)
+
+    def test__get_subject_for_value(self):
```

Comment:
Add tests for the other strategies as well.

Reply:
I will do it

Reply:
Is tests in https://github.com/confluentinc/confluent-kafka-python/pull/764/files#diff-057a19a5cfd9b6c25d4809ce6e12cf62 not enough?

Reply:
Not when you mutate the subject name outside of the subject name strategy function as you do now(appending the message field suffix in the message serializer). I'm more concerned with the final subject name than the intermediate representation. In the end my schema will be registered under this namespace so this is where we want to test for regressions.